components/auth-nav.tsx

```diff
@@ -1,41 +1,64 @@
 "use client"
 
-import React from "react"
+import React, { useCallback, useEffect, useState } from "react"
 import Link from "next/link"
 import { useRouter } from "next/navigation"
 import {
   createClientComponentClient,
   type Session,
 } from "@supabase/auth-helpers-nextjs"
 
-import { cn } from "@/lib/utils"
+import { ProfileState } from "@/types/users"
+import { cn, getCountryFlag } from "@/lib/utils"
 import { Button, buttonVariants } from "@/components/ui/button"
 
+import { Skeleton } from "./ui/skeleton"
+
 export default function AuthNav({ session }: { session: Session | null }) {
   const router = useRouter()
   const supabase = createClientComponentClient()
+  const [detail, setDetail] = useState<ProfileState>()
 
-  const handleSignOut = async () => {
+  const handleSignOut = useCallback(async () => {
     await supabase.auth.signOut()
+    setDetail(undefined)
     router.refresh()
-  }
+  }, [router, supabase])
+
+  useEffect(() => {
+    const loadProfile = async (id: string | undefined) => {
```

Comment:
Maybe use a try/catch to handle any errors:

```ts
const loadProfile = async (id: string | undefined) => {
  try {
    const userProfile = await fetch(`/api/user/profile?id=${id}`)
    const { data: profile } = await userProfile.json()
    setDetail(profile)
  } catch (error) {
    console.error('Error loading profile:', error)
  }
}
```

Reply:
The `try/catch` it's fine for me, but the additional array dependencies above it create a nightmare! :smile: